File: 04_session/Runner.scala
object StringUtils {
  def विलोम(str: String) = str.reverse
  def isPalindrome(str: String) = विलोम(str).equals(str)
}

import StringUtils._

var str = "hello";
println(विलोम(str)); // ol
println(isPalindrome(str))
println(isPalindrome("नमन"))
println("DONE")

File: 04_session/Runner.java
class StringUtils {
  
  private StringUtils() {
    
  }
  
  public static String विलोम(String str) {
    var reversed = new StringBuilder();
    for (int i = str.length() - 1; i >= 0; i--) {
      reversed.append(str.charAt(i));
    }
    return reversed.toString();
  }
  
  public static boolean isPalindrome(String str) {
    return विलोम(str).equals(str);
  }
}

public class Runner {
  public static void main(String[] args) {
    // var str = "ABLE WAS I ERE I SAW ELBA";
    var str = "hello";
    System.out.println(StringUtils.विलोम(str)); // olleh
    System.out.println(StringUtils.isPalindrome(str));  
    System.out.println(StringUtils.isPalindrome("नमन"));  
    System.out.println("DONE");
  }
}